Sources/Features/Client/Model/AttachmentsStorage.swift

```diff
@@ -0,0 +1,129 @@
+            self.fileUrl = directoryUrl.appendingPathComponent("\(fileName)_attachments.plist")
+        }
+    }
+    
+    private static let directoryName = Bundle(for: AttachmentsStorage.self).bundleIdentifier ?? "BacktraceCache"
+    
+    static func store(_ attachments: Attachments, fileName: String) throws {
+        let config = try Config(fileName: fileName)
+        
+        if !FileManager.default.fileExists(atPath: config.directoryUrl.path) {
+            try FileManager.default.createDirectory(atPath: config.directoryUrl.path,
+                                                    withIntermediateDirectories: false,
+                                                    attributes: nil)
+        }
+        
+        let attachmentBookmarks = try convertAttachmentUrlsToBookmarks(attachments)
+        
+        if #available(iOS 11.0, tvOS 11.0, macOS 10.13, *) {
+            try (attachmentBookmarks as NSDictionary).write(to: config.fileUrl)
+        } else {
+            guard (attachmentBookmarks as NSDictionary).write(to: config.fileUrl, atomically: true) else {
+                throw FileError.fileNotWritten
+            }
+        }
+        BacktraceLogger.debug("Stored attachments at path: \(config.fileUrl)")
+    }
+    
+    static func retrieve(fileName: String) throws -> Attachments {
+        let config = try Config(fileName: fileName)
+        guard FileManager.default.fileExists(atPath: config.fileUrl.path) else {
+            throw FileError.fileNotExists
+        }
+        // load file to NSDictionary
+        let dictionary: NSDictionary
+        if #available(iOS 11.0, tvOS 11.0, macOS 10.13, *) {
+            dictionary = try NSDictionary(contentsOf: config.fileUrl, error: ())
+        } else {
+            guard let dictionaryFromFile = NSDictionary(contentsOf: config.fileUrl) else {
+                throw FileError.invalidPropertyList
+            }
+            dictionary = dictionaryFromFile
+        }
+        
+        guard let bookmarks = dictionary as? Bookmarks else {
+            BacktraceLogger.debug("Could not convert stored dictionary to Bookmarks type")
+            throw FileError.unsupportedScheme
+        }
+        guard let attachments = try? extractAttachmentUrls(bookmarks) else {
+            BacktraceLogger.debug("Could not extract attachment URLs from stored attachments Bookmarks")
+            throw FileError.invalidPropertyList
+        }
+
+        BacktraceLogger.debug("Retrieved attachments from path: \(config.fileUrl)")
+        return attachments
+    }
+    
+    static func retrieveAsArray(fileName: String) throws -> [String] {
+        guard let attachments = try? retrieve(fileName: fileName) else {
+            return []
+        }
+        var attachmentsArray = [ String ]()
+        for attachment in attachments {
+            attachmentsArray.append(attachment.value.path)
+        }
+        return attachmentsArray
```

Comment:
Saw it in other places as well. You might want to think of adding a method or a computed property to get this value 

```suggestion
let attachmentsArray = attachments.map(\.value.path)
```

Reply:
I think I can remove this function, it was only used in tests anyway. Thank you for showing me how to use `map` function, this is very helpful for writing clean Swift code.